application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -122,18 +126,33 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
     }
 
     private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
-        Consumer<Throwable> handleFailure = error -> {
-            if (error instanceof ErrorResponseException) {
-                logger.warn("Unknown error occurred during help thread auto archive routine",
-                        error);
-            }
-        };
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithMention =
+                member -> threadChannel.sendMessage(member.get().getAsMention()).addEmbeds(embed);
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithoutMention =
+                member -> threadChannel.sendMessageEmbeds(embed);
 
         threadChannel.getGuild()
             .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
-            .flatMap(any -> threadChannel.getManager().setArchived(true))
-            .queue(any -> {
-            }, handleFailure);
+            .mapToResult()
+            .flatMap(member -> {
+                if (member.isSuccess()) {
+                    return sendEmbedWithMention.apply(member);
+                }
+
+                logger.debug("Unable to mention user", member.getFailure());
```

Comment:
As discussed on the server already, we should add more context to this. I would consider this a `warning` but if this happens all the time, can we change this bit of code so it's clear what's going on e.g. adding a check so see if the user is in server, if they are, error/warn log this message. Otherwise, info log that they have left.

Reply:
Also, constants should be uppercase. As should the `Logger`, so make it `LOGGER` while you're here please. It's not ugly to make it LOGGER.

Reply:
since we already handled this with `sendEmbedWithoutMention`, it just had to be there could be in debug or info as long as it's taken care of.

Reply:
The problem is that you're needing to explain this. At the minimum add documentation to anybody else without context will know what's going on. But I'd much rather prefer the code to be rewritten in a way that can express it. You can choose the solution here either way.

Reply:
I have made some changes in pipeline to display intent more clearly, lmk what you think.

Reply:
this is valid but everywhere else in the code, it's `logger`, so for consistency, keep it `logger`. And if you want to  change it,  change it everywhere else as well in a separate pr

Reply:
i was wondering why i used `logger` twice in both PRs, prolly followed from rest of code. Can we start from here? It prolly should be done anyway.

Reply:
If it's lowercase logger everywhere then yeah it should be changed. Surprised the linter doesn't flag it. But changing it everywhere is out of the scope of this PR. 

We have 2 solutions:

1. Create a PR to address this 
2. Update whenever we touch a file

I'd go for the second option, if we see such cases, then they can be fixed when next touched. 

We can raise a request for somebody to do it in a big PR though for those who want to contribute but don't understand enough. 

In any case, it's good how it is now in this PR with the uppcase.

Reply:
> In any case, it's good how it is now in this PR with the uppcase.

its not, no one has agreed to the change + it'll only make the code more inconsistent

Reply:
changed back to `logger`, new PR can be raised for refactoring later.